.github/workflows/release.yml

```diff
@@ -0,0 +1,92 @@
+    - "v[0-9]+.[0-9]+.[0-9]+"
+    # match semver pre-releases
+    - "v[0-9]+.[0-9]+.[0-9]+-*"
+jobs:
+  release:
+    env:
+      NETLIFY_BASE: 'videojs-preview.netlify.app/sandbox/'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        # We neeed to fetch the entire history as conventional-changelog needs
+        # access to any number of git commits to build the changelog.
+        with:
+          fetch-depth: 0
+
+      - name: read node version from .nvmrc
+        run: echo ::set-output name=NVMRC::$(cat .nvmrc)
+        shell: bash
+        id: nvm
+
+      - name: setup node
+        uses: actions/setup-node@v2
+        with:
+          node-version: '${{steps.nvm.outputs.NVMRC}}'
+          cache: npm
+          registry-url: 'https://registry.npmjs.org'
+
+      - name: npm install
+        run: npm i --prefer-offline --no-audit
+
+      # publish runs build for us via a prepublishOnly script
+      - name: npm release
+        run: npm publish --tag next
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+
+      - name: get current changelog
+        run: node -e "console.log(require('./build/current-changelog.js')())" > CHANGELOG-LATEST.md
+
+      - name: get dashed package version for netlify
+        run: echo ::set-output name=VERSION::$(node -e "process.stdout.write(require('./package.json').version.split('.').join('-'))")
+        id: get-version
+        shell: bash
+        if: env.NETLIFY_BASE != ''
+
+      - name: add netlify preview to release notes
+        run: |
+          echo "" >> CHANGELOG-LATEST.md
+          echo "[netlify preview for this version](https://v${{steps.get-version.outputs.VERSION}}--${{env.NETLIFY_BASE}})" >> CHANGELOG-LATEST.md
+        if: env.NETLIFY_BASE != ''
+
+      - name: Create Github release
+        uses: softprops/action-gh-release@v1
+        with:
+          body_path: CHANGELOG-LATEST.md
+          token: ${{github.token}}
+          prerelease: true
+          files: dist/*.zip
+          discussion_category_name: Releases
+
+  deploy:
+    needs: release
+    runs-on: ubuntu-latest
+    environment: Deploy
+    env:
+      VJS_VERSION: ${{ github.event.head_commit.message }}
+      S3_BUCKET: ${{ secrets.AWS_S3_BUCKET }}
+      S3_ACCESS: ${{ secrets.AWS_S3_ACCESS }}
+      S3_KEY: ${{ secrets.AWS_S3_KEY }}
+    steps:
+      - name: setup node
+        uses: actions/setup-node@v2
+
+      - name: npm install video.js
+        run: npm install "video.js@$VJS_VERSION"
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_S3_SECRET }}
+          aws-region: us-east-1
+
+      - name: Copy files to the CDN with the AWS CLI
+        run: |
+          aws s3 sync node_modules/video.js/dist/ s3://${S3_KEY}/${VJS_VERSION}/ --acl ${S3_ACCESS}
```

Comment:
this should upload to a full version specific folder, i.e. `7.15.0/video.js`. I may create a `promote` workflow that can set the npm tag to latest and also upload it to s3 under the minor, i.e. `7.15/video.js` (and maybe major too, i.e. `7/video.js`).

Reply:
While this should work, I wasn't able to test it directly, so, our first release with this will likely be also our test. I was able to test the gh and npm release in a test project and those worked flawlessly.

Reply:
Creating a new task would be cool, because we could potentially automate updating the GH Release to no longer be a pre-release either.